specs.md

```diff
@@ -19,50 +19,140 @@ This approach is based on data gathered from public automatically resetted netwo
 
-At the very beginning of testnet existence, in the first iteration, `period 0`, the network starts with `genesis 0`. This very fist genesis is hardcoded in the client like any other predefined network. However, during each client start, client does not initiatilize this genesis but rather builds a new one based on it. This new genesis, derived from the `genesis_0`, will be written to the database and used to run the current network. 
+The main changes in the genesis iteration are chainId, timestamp and the withdrawal credentials of the first validator.
+    
+The network always starts from a genesis which is deterministically created based on the original one, genesis_0. Expiration of the genesis is given by its `genesis_timestamp` and `period` (a predefined constant, length of time a single ephemeral network runs). Therfore once the currenttimestamp reaches the terminal time of ephemeral network, it has to switch to a new genesis. This can be expressed as:
 
-Given a known `period` (the length of time network instance will run), the client can calculate the number of lifecycle iterations from `genesis_0` to the current one and create a new genesis with latest parameters. Always, when the client starts with an option of ephemeral testnet, it checks the current genesis timestamp and identifies whether it's older than `timestamp + period`, at which point it triggers generation of a new genesis.
+    let terminal_timestamp = genesis_timestamp + period
+    
+    if (current_timestamp >= terminal_timestamp):
+       return new_genesis
+    else:
+       return none
 
-New genesis needs to be created and correspond in both EL and CL client. Manual implementation of this mechanism can be found in [this script](https://hackmd.io/@jimmygchen/rJw2Zjl2j). 
+Clients shall include a hardcoded genesis_0, similarly to other predefined networks. But this genesis is only used at the very beginning of the testnet existence, in its first iteration 0. Later on, with iteration 1 and further, client does not initialize this genesis but uses it to derive the current one. 
+
+Thus, for any iteration i, after the first iteration (i > 0):
+
+    genesis_timestamp_i = period * i + genesis_timestamp_0.
+
+When the client starts with the option of ephemeral testnet, it checks whether a genesis for the network is present. If it doesn't exist or the current genesis timestamp is older than genesis_timestamp + period, it triggers the generation of a new genesis. This new genesis, derived from the genesis 0, will be written to the database and used to run the current network.
+
+      let genesis = genesis_from_database()
+      
+      if (genesis = 0 || current_timestamp >= min_genesis_time + period):
+         generate new_genesis 
+         return genesis_from_database(new_genesis)
+      else:
+         return genesis_from_database()
+
+The choice of period can be very important and have certain implications on netowrk stability and overall user experience. 
```

Comment:
The period reasoning is explained below, you can remove this part or add it below